client/ayon_core/hosts/maya/api/testing/tests.py

```diff
@@ -0,0 +1,197 @@
+import traceback
+
+import maya.cmds as cmds
+
+import pyblish.util
+
+from ayon_core.pipeline import (
+    registered_host,
+    get_current_folder_path,
+    get_current_task_name
+)
+from ayon_core.pipeline.create import CreateContext
+from ayon_core.hosts.maya.api.workfile_template_builder import (
+    MayaTemplateBuilder
+)
+from ayon_core.hosts.maya.api.lib import set_attribute
+
+from . import lib
+
+
+# Needed for transition phase for asset/subset renaming. Can be hardcoded once
+# transition is done.
+product_key_name = "productName"
+product_type_key_name = "productType"
+
+
+def test_create():
```

Comment:
Not entirely sure what you mean, but we are going through the creators `create` method.

We are querying the members of the sets, since the selection when creating the instances are not recording elsewhere.

Reply:
I wonder why we're testing "recreating" instances? Shouldn't we just check whether a regular "create" works?

Reply:
Anyway, I meant going through `CreateContext.create` instead of `Creator.create`

Like:
```python
created_instance = context.create(
    creator_identifier=creator_identifier,
    variant=variant,
    pre_create_data={"use_selection": True}
)
```

Similarly, I would set the creator attributes and publish attribute values via the created instance:

```python
for key, value in old_instance.creator_attributes:
    created_instance.creator_attributes[key] = value
```

And then allowing the CreateContext and `update_instances` logic of the `Creators` to trigger and ensure the saving of the instances logic works as intended.

```python
context.save_changes()
```

Making something like:

```python
created_instance = context.create(
    creator_identifier=creator_identifier,
    variant=variant,
    pre_create_data={"use_selection": True}
)

for key, value in old_instance.creator_attributes:
    created_instance.creator_attributes[key] = value

context.save_changes()
```

---

Also, pretty sure we want `use_selection` to be `False` since we apparently seem to be forcing members into the created instances to match the original instance's members?

Reply:
Cool, this all sounds good.